syntax = "proto3";
import "google/protobuf/empty.proto";
package pb;

option go_package = "./pb";


message ProductInteraction {
    string user_id = 1;
    string product_id = 2;
    string interaction_type = 3;
}

message RecommendationRequestForUserId {
    string user_id = 1;
    uint64 skip = 2;
    uint64 take = 3;
}

message RecommendationRequestOnViews {
    repeated string ids = 1;
    uint64 skip = 2;
    uint64 take = 3;
}

message ProductReplica {
    string id = 1;
    string name = 2;
    string description = 3;
    double price = 4;
}

message RecommendationResponse {
    repeated ProductReplica recommended_products = 1;
}

service RecommenderService {
    rpc GetRecommendationsForUserId (RecommendationRequestForUserId) returns (RecommendationResponse);
    rpc GetRecommendationsOnViews (RecommendationRequestOnViews) returns (RecommendationResponse);
    rpc RecordInteraction (ProductInteraction) returns (google.protobuf.Empty);
}


